compiler/rustc_const_eval/src/const_eval/valtrees.rs

```diff
@@ -152,6 +152,8 @@ pub(crate) fn const_to_valtree_inner<'tcx>(
         // FIXME(oli-obk): we can probably encode closures just like structs
         | ty::Closure(..)
         | ty::Generator(..)
+        // FIXME: we could look behind type aliases.
+        | ty::TyAlias(..)
```

Comment:
These should be unreachable. If we computed a layout, we already looked behind type aliases. We could move this, as well as `Param`, `Bound`, `Projection`, `Infer` and `PlaceHolder` to a separate arm that just does a `bug!()` call